File: LearnSparkJvm/src/main/java/com/dt/learn/rdd/CreateRdd.java
package com.dt.learn.rdd;

import com.dt.learn.SparkContextApplication;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

/**
 * @author dt 2018/7/15 20:04
 */
public class CreateRdd {

    /**
     * 使用简单的方式创建RDD
     * @return
     */
    public JavaRDD<Integer> createRddForSimple(){

        JavaSparkContext javaSparkContext = SparkContextApplication.crateSparkContextFactory();
        JavaRDD<Integer> rdd = javaSparkContext.parallelize(Arrays.asList(1, 2, 3, 4, 5, 6, 7));
        return rdd;

    }

}

File: LearnSparkJvm/src/main/scala/com/dt/learn/word/rdd/CreateRdd.scala
package com.dt.learn.word.rdd

import com.dt.learn.SparkContextScalaApplication
import org.apache.spark.rdd.RDD

/**
  * @author dt 2018/7/15 19:49
  */
object CreateRdd {

  /**
    * 通过parallelize创建一个简单的RDD
    * @return
    */
  def createRddForSimple():RDD[Int]={
    val sc = SparkContextScalaApplication.createSparkContextForConf
    val rdd = sc.parallelize(List(1,2,3,4,5,6,7))
    rdd
  }

}
